rts/motoko-rts/src/component.rs

```diff
@@ -0,0 +1,56 @@
+// Wasm Component Model utilities
+
+use crate::{
+    barriers::allocation_barrier,
+    memory::{alloc_blob, Memory},
+    types::{Bytes, Value, TAG_BLOB_B},
+};
+use alloc as alloc_crate;
+use core::convert::TryInto;
+use motoko_rts_macros::ic_mem_fn;
+
+///  Convert a Canonical ABI `list<u8>` pointer to a Motoko `Blob`.
+#[ic_mem_fn]
+unsafe fn blob_of_cabi<M: Memory>(mem: &mut M, ret: *const usize) -> Value {
+    let content = *ret as *const u8;
+    let len = *ret.add(1);
+
+    // TODO: reuse Blob from `cabi_realloc`?
```

Comment:
no need for that, `cabi_realloc` is only needed to allocate memory for the rust component.
Here we do the inverse, allocate for Motoko